templates/_macros/hero.jinja

```diff
@@ -0,0 +1,127 @@
+# Params
+# title_text: Hero title text (required)
+# subtitle_text: Hero subtitle text
+# layout: layout of hero section. Options are '50/50', '75/25', '25/75', 'fallback'
+# is_split_on_tablet: whether the layout is split on tablet. Applies "-on-medium" to row class in 50/50 and 25/75 layouts. If false, the layout is stacked on tablet.
+# is_split_on_desktop: whether the layout is split on desktop. Applies "-on-large" to row class in 50/50 and 25/75 layouts. If false, the layout is stacked on desktop, provided is_split_on_tablet is true.
+# is_image_full_cover: whether the image should span full width of the hero. Only supported on 50/50 layout
+# cta_primary_url: primary CTA button URL
+# cta_secondary_url: secondary CTA button URL
+# Slots
+# description: paragraph-style content below the title and subtitle
+# cta_primary: contents of the primary CTA button
+# cta_secondary: contents of the secondary CTA button
+# image: slot for image content (required)
+
+{% macro hero(
+  title_text='',
+  subtitle_text='',
+  layout="fallback",
+  is_split_on_tablet = false,
+  is_split_on_desktop = true,
+  is_image_full_cover = false,
+  cta_primary_url = '#',
+  cta_secondary_url = '#'
+) -%}
+  {% set has_subtitle = subtitle_text|trim|length > 0 %}
+  {% set description_content = caller('description') %}
+  {% set has_description = description_content|trim|length > 0 %}
+  {% set cta_primary_content = caller('cta_primary') %}
+  {% set cta_secondary_content = caller('cta_secondary') %}
+  {% set has_cta = cta_primary_content|trim|length > 0 %}
+  {% set has_cta_secondary = cta_secondary_content|trim|length > 0 %}
+  {% set image_content = caller('image') %}
+
+  {# User can pass layout as "X-Y" or "X/Y" #}
+  {% set layout = layout | trim | replace('/', '-') %}
+  {% if layout not in ['50-50', '75-25', '25-75', 'fallback'] %}
+    {% set layout = 'fallback' %}
+  {% endif %}
+
+  {# Full cover image only supported on 50-50 layout #}
+  {% set is_image_full_cover = is_image_full_cover and layout == '50-50' %}
+
+  {# Construct grid classes from params input #}
+  {% set row_classes = [] %}
+  {% set col_classes = ['col', 'col'] %}
+  {% if layout in ['50-50', '25-75'] %}
+    {# Responsive breakpoint classes were requested #}
+    {% if is_split_on_tablet or is_split_on_desktop %}
+      {% if is_split_on_tablet %}
+        {{ row_classes.append('row--' + layout + '-on-medium') or "" }}
+      {% endif %}
+      {% if is_split_on_desktop %}
+        {{ row_classes.append('row--' + layout + '-on-large') or "" }}
+      {% endif %}
+    {% else %}
+      {{ row_classes.append('row--' + layout) or "" }}
+    {% endif %}
+  {% elif layout  == '75-25' %}
+    {# there is no .row--75-25 class, so we use unmodified row and col-9 and col-3 to create 75-25 layout #}
+    {{ row_classes.append('row') or "" }}
+    {% set col_classes = ['col-9', 'col-3'] %}
```

Comment:
If the lack of `row--75-25` is making this more complicated, lets add it.

Reply:
Working on this today as #5234

Reply:
Removed the 75-25 code from this PR entirely, it'll be re-introduced in 75-25 PR